CMakeLists.txt

```diff
@@ -0,0 +1,64 @@
+cmake_minimum_required(VERSION 3.4)
+project(RdKafka)
+
+# Options. No 'RDKAFKA_' prefix to match old C++ code. {
+
+# This option doesn't affect build in fact, only C code
+# (see 'rd_kafka_version_str'). In CMake the build type feature usually used
+# (like Debug, Release, etc.).
+option(WITHOUT_OPTIMIZATION "Disable optimization" OFF)
+
+option(ENABLE_DEVEL "Enable development asserts, checks, etc" OFF)
+option(ENABLE_REFCNT_DEBUG "Enable refcnt debugging" OFF)
+option(ENABLE_SHAREDPTR_DEBUG "Enable sharedptr debugging" OFF)
+
+option(WITH_ZLIB "With ZLIB" ON)
```

Comment:
The prefix might be unfortunate in librdkafka, but:
 * `WITH_foo` = foo was found and support for it will be compiled in (e.g., zlib compression)
 * `ENABLE_foo` = --enable-foo was specified to ./configure and we should try to set `WITH_foo` if the library is found. (this is mostly used in the reverse way, to --disable-ssl certain features).

This is probable the other way around in CMake, but we'll need to ignore that fact now since the rest of the code relies on the above assumptions.

Also: I think it is okay for some features to be left out if dependent libraries are not found, and that's how mklove and autoconf works today, so I dont think we should change that approach when cmake is used.